lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -178,12 +178,19 @@ protected function _getPortableTableColumnDefinition($tableColumn)
 
         $length = ((int) $length == 0) ? null : (int) $length;
 
+        $isNotNull = ($tableColumn['null'] !== 'YES');
+        $columnDefault = (isset($tableColumn['default'])) ? $tableColumn['default'] : null;
+        if ($columnDefault === 'NULL' && !$isNotNull) {
```

Comment:
This looks a bit weird. How does MariaDB 10.2.7 distinct between default `NULL` and '`NULL'`?

Reply:
Yes I was wondering too... looks they tried to fix something else and got a regression. Possible they will revert this... in this case we could eventually check the server version, and apply the fix selectively for a while.

Reply:
They are not going to revert it... they are planing to enforce it as a standard in all upcoming versions, see [https://jira.mariadb.org/browse/MDEV-13341](https://jira.mariadb.org/browse/MDEV-13341)
Quoted from above link:

> It was an intentional change...

> We've tried to reduce the damage by doing this change only in 10.2 (not in 10.1 or 10.0 or 5.5) and by keeping old behavior of SHOW COLUMNS (because it's non-standard, so there're no rules for it). Applications can switch to SHOW COLUMNS easily. Or adapt to use the new behavior, because old one cannot describe default values unambiguously.

Reply:
Was wondering, but now I've tested and figured out there's no problem at all. To the question of @morozov , how to disambiguate ? actually very simple:

```sql
CREATE TABLE `my_table` ( 
    `column_1` VARCHAR(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT NULL , 
    `column_2` VARCHAR(50) CHARACTER SET utf8 COLLATE utf8_general_ci NULL DEFAULT 'NULL' 
) ENGINE = InnoDB;
```
If you read the `information_schema`, like

```sql
SELECT * FROM `COLUMNS` where TABLE_SCHEMA = 'mytest'
```

You'll see defaults are `NULL` and `'NULL'`. So no problem.

Reply:
@belgattitude my question was about the information schema, not about the actual table. What does it contain for the columns above?